libs/core/synchronization/include/hpx/synchronization/condition_variable.hpp

```diff
@@ -43,43 +105,109 @@ namespace hpx {
           : data_(data_type(
                 new lcos::local::detail::condition_variable_data, false))
         {
         }
 
-        // Preconditions: There is no thread blocked on *this. [Note: That is,
-        //      all threads have been notified; they could subsequently block
-        //      on the lock specified in the wait.This relaxes the usual rules,
-        //      which would have required all wait calls to happen before
-        //      destruction.Only the notification to unblock the wait needs to
-        //      happen before destruction.The user should take care to ensure
-        //      that no threads wait on *this once the destructor has been
-        //      started, especially when the waiting threads are calling the
-        //      wait functions in a loop or using the overloads of wait,
-        //      wait_for, or wait_until that take a predicate. end note]
-        //
-        // IOW, ~condition_variable() can execute before a signaled thread
-        // returns from a wait. If this happens with condition_variable, that
-        // waiting thread will attempt to lock the destructed mutex.
-        // To fix this, there must be shared ownership of the data members
-        // between the condition_variable_any object and the member functions
-        // wait (wait_for, etc.).
+        /**
+         * \brief Destroys the object of type \a hpx::condition_variable. 
+         * 
+         * \note Preconditions: There is no thread blocked on *this. [Note:
+         *                      That is, all threads have been notified; they
+         *                      could subsequently block on the lock specified
+         *                      in the wait.This relaxes the usual rules,
+         *                      which would have required all wait calls to
+         *                      happen before destruction. Only the
+         *                      notification to unblock the wait needs to
+         *                      happen before destruction.The user should take
+         *                      care to ensure that no threads wait on *this
+         *                      once the destructor has been started,
+         *                      especially when the waiting threads are calling
+         *                      the \a wait functions in a loop or using the
+         *                      overloads of \a wait, \a wait_for, or
+         *                      \a wait_until that take a predicate. end note]
+         *
+         * IOW, \a ~condition_variable() can execute before a signaled thread
+         * returns from a wait. If this happens with \a condition_variable,
+         * that waiting thread will attempt to lock the destructed mutex.
+         * To fix this, there must be shared ownership of the data members
+         * FIXME(bhumit->hkaiser) shouldn't this be condition_variable_any?
+         * between the \a condition_variable_any object and the member
```

Comment:
@hkaiser shouldn't this be `condition_variable_any`? I have a feeling I'm wrong, however

Reply:
This should be `condition_variable`, I think.